applications/GeoMechanicsApplication/tests/cpp_tests/test_reset_displacement_process.cpp

```diff
@@ -0,0 +1,158 @@
+                                                          model_part.GetProcessInfo());
+
+    KRATOS_EXPECT_EQ(constitutive_laws.size(), number_of_integration_points);
+    for (const auto& constitutive_law : constitutive_laws) {
+        KRATOS_EXPECT_VECTOR_NEAR(constitutive_law->GetInitialState().GetInitialStressVector(),
+                                  initial_stress_vector, 1e-12)
+    }
+}
+
+KRATOS_TEST_CASE_IN_SUITE(ResetDisplacementProcess_ThrowsInCheck_WhenModelIsNotRestarted,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    Model model;
+    auto& model_part                          = model.CreateModelPart("Main");
+    model_part.GetProcessInfo()[IS_RESTARTED] = false;
+
+    ResetDisplacementProcess process(model_part, {});
+
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(
+        process.Check(), "The IS_RESTARTED flag must be set to true in the ProcessInfo of the "
+                         "model part. Please use the \"rest\" option for the model input type")
+
+    model_part.GetProcessInfo()[IS_RESTARTED] = true;
+    KRATOS_EXPECT_EQ(process.Check(), 0);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(ResetDisplacementProcess_ThrowsInExecuteInitialize_WhenConstitutiveLawsCannotBeRetrieved,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    Model model;
+    auto& model_part = CreateModelPartWithAStubElement(model);
+
+    ResetDisplacementProcess process(model_part, {});
+    dynamic_cast<StubElement&>(model_part.Elements()[1]).mConstitutiveLaws.clear();
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(
+        process.ExecuteInitialize(),
+        "The constitutive laws on the integration points could not be retrieved for element 1")
+}
+
+KRATOS_TEST_CASE_IN_SUITE(ResetDisplacementProcess_ThrowsInExecuteInitialize_WhenStressVectorsCannotBeRetrieved,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    Model model;
+    auto& model_part = CreateModelPartWithAStubElement(model);
+
+    ResetDisplacementProcess process(model_part, {});
+
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(
+        process.ExecuteInitialize(),
+        "The stress vectors on the integration points could not be retrieved for element 1")
+}
+
+KRATOS_TEST_CASE_IN_SUITE(ResetDisplacementProcess_ThrowsInExecuteInitialize_WhenStressVectorsAndConstitutiveLawsAreNotTheSameSize,
+                          KratosGeoMechanicsFastSuiteWithoutKernel)
+{
+    Model model;
+    auto& model_part = CreateModelPartWithAStubElement(model);
+    dynamic_cast<StubElement&>(model_part.Elements()[1]).mIntegrationPointVectors =
+        std::vector<Vector>(2, ScalarVector(4, 1.0));
+
+    ResetDisplacementProcess process(model_part, {});
+
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(process.ExecuteInitialize(),
+                                      "Number of retrieved stress vectors does not match the "
+                                      "number of constitutive laws for element 1")
+}
```

Comment:
When finding out what is wrong from error messages I'm often happy if information about the numbers is given, such that I have an indication which of the 2 is the wrong one.

Reply:
Good suggestion, added!